upcoming_changes/3238.api

```diff
@@ -0,0 +1,5 @@
+Improve the readibility fo the code by replacing the ``__call__`` method of some objects with the more explicit ``_get_current_data``.
```

Comment:
It may be worth mentioned explicitly that `s()`  where `s` being an instance of `BaseSignal` is removed. Most users will not know that `s()` is the same as `s.__call__`.
As this privatise this functionality, it may be worth mentioned it explicitly here?